zbus/src/connection/mod.rs

```diff
@@ -1370,6 +1402,112 @@ mod tests {
         let name_has_owner = dbus.name_has_owner(name.try_into().unwrap()).await.unwrap();
         assert!(!name_has_owner);
     }
+
+    #[tokio::test(start_paused = true)]
+    #[timeout(15000)]
+    async fn test_graceful_shutdown() {
+        // If we have the only reference, it should complete immediately.
+        let connection = Connection::session().await.unwrap();
+        let shutdown = connection.graceful_shutdown();
+        tokio::select! {
+            biased;
+            _ = shutdown => {},
+            _ = std::future::ready(()) => {
+                panic!("Graceful shutdown did not immediately complete");
```

Comment:
I wonder if this could be potentially problematic. If we receive any method call (e.g a generic client like d-feet running on the bus and calling introspect on all new peers) in the very short lifetime of this connection. :thinking: I don't think we need to test this even since it's not the problematic part. If people want immediate shutdown, they can always `drop` the connection instead.